test/e2e/self_hosted.go

```diff
@@ -89,11 +86,11 @@ func SelfHostedSpec(ctx context.Context, inputGetter func() SelfHostedSpecInput)
 				Flavor:                   clusterctl.DefaultFlavor,
 				Namespace:                namespace.Name,
 				ClusterName:              fmt.Sprintf("cluster-%s", util.RandomString(6)),
-				KubernetesVersion:        input.GetMinimumCAPISupportedKubernetesVersion(),
+				KubernetesVersion:        input.E2EConfig.GetVariable(MinimumCAPISupportedKubernetesVersion),
```

Comment:
Might be I missed something, but why in some test we use the minimum supported version and in some other current version? 
IMO PR pre-submit job should always stay on top of things (master/current); if tests for older version are required, we can eventually create periodic tests (or more pre-submit)

Reply:
So this is a change introduced in this PR: https://github.com/kubernetes-sigs/cluster-api/pull/2968
@randomvariable is there a particular reason for using minimum supported version in this spec?